src/test/java/KeyProviderTest.java

```diff
@@ -26,32 +30,52 @@
   private static final String SAMPLE_KEY = "sampleKey";
   private static final String CAPITALISED_SAMPLE_KEY = "SAMPLEKEY";
   private static final String INVALID_KEY = "invalidKey";
-
   private static final String SAMPLE_VALUE = "sampleValue";
+  private static final String KEYS_JSON = "{\"sampleKey\" : \"sampleValue\"}";
+  private static final File file = new File("src/main/resources/TEST_KEYS.json");
+
+  @Before
+  public void init() {
+    // Creates a new file in src/main/resources and writes json content to the file.
+    try {
+      file.getParentFile().mkdirs();
+      file.createNewFile();
+      FileWriter writer = new FileWriter("src/main/resources/TEST_KEYS.json");
+      writer.write(KEYS_JSON);
+      writer.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @After
+  public void tear() {
```

Comment:
For similar reasons as above, try to use `tearDown` for `@After` functions

Reply:
Thanks for letting me know, I'll be sure to keep that in mind going forward!